lib/transactions.js

```diff
@@ -122,13 +138,30 @@ class Transaction {
     const nextStates = stateMachine[this.state];
     if (nextStates && nextStates.indexOf(nextState) !== -1) {
       this.state = nextState;
+      if (this.state === TxnState.NO_TRANSACTION || this.state === TxnState.STARTING_TRANSACTION) {
+        this.unpinServer();
+      }
       return;
     }
 
     throw new MongoError(
       `Attempted illegal state transition from [${this.state}] to [${nextState}]`
     );
   }
+
+  get isPinned() {
```

Comment:
it looks like `get isPinned` is defined twice here, and not actually used.

Reply:
whoops, removing